File: xv/patch/XV3080Voice.swift

extension XV3080 {
  
  enum Voice {
    
    static let patchWerk = XV.Voice.patchWerk(common: Common.patchWerk, tone: Tone.patchWerk, fx: FX.patchWerk, chorus: Chorus.patchWerk, reverb: Reverb.patchWerk, initFile: "xv5050-voice-init")
    
    static let bankWerk = XV.Voice.bankWerk(patchWerk)
    
    enum Common {
      static let patchWerk = XV.Voice.Common.patchWerk(params)
      
      static let parms = XV5050.Voice.Common.parms + [
        .p([.out, .assign], 0x27, .options(outAssignOptions))
      ]
      static let params = parms.params()
      
      static let outAssignOptions = [
        0 : "MFX",
        1 : "A",
        2 : "B",
        3 : "C",
        5 : "1",
        6 : "2",
        7 : "3",
        8 : "4",
        9 : "5",
        10 : "6",
        13 : "Tone",
        ]
    }
    
    enum Tone {
      
      static let patchWerk = XV.Voice.Tone.patchWerk(params: params)
      
      static let parms = XV5050.Voice.Tone.parms + [
        .p([.out, .assign], 0x11, .options(outAssignOptions))
      ]
      static let params = parms.params()

      static let outAssignOptions = [
        0 : "MFX",
        1 : "A",
        2 : "B",
        3 : "C",
        5 : "1",
        6 : "2",
        7 : "3",
        8 : "4",
        9 : "5",
        10 : "6",
        ]
      
      public static let waveGroupOptions =
        XV5050.Voice.Tone.internalWaveGroupOptions <<<
        XV5050.Voice.Tone.srxWaveGroupOptions <<<
        XV5050.Voice.Tone.srjvWaveGroupOptions
    }
  }
}

File: xv/xv3080_voice.js

const commonOutAssignOptions = Array.sparse([
  [0, "MFX"],
  [1, "A"],
  [2, "B"],
  [3, "C"],
  [5, "1"],
  [6, "2"],
  [7, "3"],
  [8, "4"],
  [9, "5"],
  [10, "6"],
  [13, "Tone"],
])

const commonParms = XV5050Voice.commonParms.concat([
  ['out/assign', { b: 0x27, opts: commonOutAssignOptions }]
])

const commonPatchWerk = XVVoice.commonPatchWerk(commonParms)

const toneOutAssignOptions = Array.sparse([
  [0, "MFX"],
  [1, "A"],
  [2, "B"],
  [3, "C"],
  [5, "1"],
  [6, "2"],
  [7, "3"],
  [8, "4"],
  [9, "5"],
  [10, "6"],
])

const toneParms = XV5050Voice.toneParms.concat([
  ['out/assign', { b: 0x11, opts: toneOutAssignOptions }],
])

const tonePatchWerk = XVVoice.tonePatchWerk(toneParms)

const waveGroupOptions =
  XV5050Voice.Tone.internalWaveGroupOptions <<<
  XV5050Voice.Tone.srxWaveGroupOptions <<<
  XV5050Voice.Tone.srjvWaveGroupOptions

const patchWerk = XV.Voice.patchWerk(common: Common.patchWerk, tone: Tone.patchWerk, fx: FX.patchWerk, chorus: Chorus.patchWerk, reverb: Reverb.patchWerk, initFile: "xv5050-voice-init")
  
const bankWerk = XV.Voice.bankWerk(patchWerk)

module.exports = {
  patchWerk,
  bankWerk,
}